signer/src/request_decider.rs

```diff
@@ -0,0 +1,435 @@
+//! # Request decider event loop
+//!
+//! This module contains the request decider, which is the component of the sBTC signer
+//! responsible for deciding whether to accept or reject a request.
+//!
+//! For more details, see the [`RequestDeciderEventLoop`] documentation.
+
+use crate::blocklist_client::BlocklistChecker;
+use crate::context::Context;
+use crate::context::P2PEvent;
+use crate::context::RequestDeciderEvent;
+use crate::context::SignerCommand;
+use crate::context::SignerEvent;
+use crate::context::SignerSignal;
+use crate::ecdsa::SignEcdsa as _;
+use crate::ecdsa::Signed;
+use crate::error::Error;
+use crate::keys::PrivateKey;
+use crate::keys::PublicKey;
+use crate::message::Payload;
+use crate::message::SignerDepositDecision;
+use crate::message::SignerMessage;
+use crate::message::SignerWithdrawalDecision;
+use crate::network::MessageTransfer;
+use crate::storage::model;
+use crate::storage::model::BitcoinBlockHash;
+use crate::storage::model::DepositSigner;
+use crate::storage::model::WithdrawalSigner;
+use crate::storage::DbRead as _;
+use crate::storage::DbWrite as _;
+
+use futures::StreamExt;
+use futures::TryStreamExt;
+
+/// This struct is responsible for deciding whether to accept or reject
+/// requests and persisting requests from other signers.
+#[derive(Debug)]
+pub struct RequestDeciderEventLoop<C, N, B> {
+    /// The signer context.
+    pub context: C,
+    /// Interface to the signer network.
+    pub network: N,
+    /// Blocklist checker.
+    pub blocklist_checker: Option<B>,
+    /// Private key of the signer for network communication.
+    pub signer_private_key: PrivateKey,
+    /// How many bitcoin blocks back from the chain tip the signer will look for requests.
+    pub context_window: u16,
+}
+
+impl<C, N, B> RequestDeciderEventLoop<C, N, B>
+where
+    C: Context,
+    N: MessageTransfer,
+    B: BlocklistChecker,
+{
+    /// Run the signer event loop
```

Comment:
nit: update comment

Reply:
Fixed in https://github.com/stacks-network/sbtc/pull/872/commits/771e5076cff87a960485fea53ff6981529e8c1b0.